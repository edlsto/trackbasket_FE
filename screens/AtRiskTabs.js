import { MaterialIcons } from 'react-native-vector-icons';
import React, { useContext } from 'react';
import { Text, View } from 'react-native';
import UserContext from '../user-context';
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Cart from './Cart';
import Shop from './Shop';

const Tab = createBottomTabNavigator();

const AtRiskTabs = () => {
  const { cart } = useContext(UserContext);
  function IconWithBadge({ name, badgeCount, color, size }) {
    return (
      <View style={{ width: 24, height: 24, margin: 5 }}>
        <MaterialIcons name={name} size={size} color={color} />
        {badgeCount > 0 && (
          <View
            style={{
              // On React Native < 0.57 overflow outside of parent will not work on Android, see https://git.io/fhLJ8
              position: 'absolute',
              right: -6,
              top: -3,
              backgroundColor: 'red',
              borderRadius: 6,
              width: 12,
              height: 12,
              justifyContent: 'center',
              alignItems: 'center',
            }}
          >
            <Text style={{ color: 'white', fontSize: 10, fontWeight: 'bold' }}>
              {badgeCount}
            </Text>
          </View>
        )}
      </View>
    );
  }

  function HomeIconWithBadge(props) {
    // You should pass down the badgeCount in some other ways like React Context API, Redux, MobX or event emitters.
    if (cart) {
      return (
        <IconWithBadge
          {...props}
          badgeCount={cart.items.reduce((itemCount, item) => {
            itemCount += item.quantity;
            return itemCount;
          }, 0)}
        />
      );
    } else {
      return '';
    }
  }

  return (
    <Tab.Navigator>
      <Tab.Screen
        component={Shop}
        name="Shop"
        options={{
          tabBarLabel: 'Shop',
          tabBarIcon: ({ color, size }) => (
            <MaterialIcons name="shopping-basket" color={color} size={size} />
          ),
        }}
      />
      <Tab.Screen
        component={Cart}
        name="Cart"
        options={{
          tabBarLabel: 'Cart',
          tabBarIcon: ({ color, size }) => (
            <HomeIconWithBadge name="shopping-cart" color={color} size={size} />
          ),
        }}
      />
    </Tab.Navigator>
  );
};

export default AtRiskTabs;
