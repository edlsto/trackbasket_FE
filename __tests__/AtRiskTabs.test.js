import React from 'react';
import AtRiskTabs from '../screens/AtRiskTabs';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <AtRiskTabs />
      </NavigationContainer>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
