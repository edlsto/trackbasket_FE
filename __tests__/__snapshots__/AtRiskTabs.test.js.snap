// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            null,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "flex": 1,
                  "paddingBottom": 100,
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 350,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Search Items..."
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "borderWidth": 1,
                      "fontSize": 18,
                      "height": 50,
                      "marginBottom": 40,
                      "marginTop": 10,
                      "padding": 5,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <RCTScrollView>
                  <View />
                </RCTScrollView>
                <Text
                  style={
                    Object {
                      "color": "grey",
                      "fontFamily": "HelveticaNeue",
                      "fontSize": 18,
                      "textAlign": "center",
                    }
                  }
                >
                  Enter a search term.
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(224, 224, 224)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 50,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="Shop, tab, 1 of 2"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(0, 122, 255)",
                "fontSize": 11,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            Shop
          </Text>
        </View>
        <View
          accessibilityLabel="Cart, tab, 2 of 2"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#8E8E8F",
                "fontSize": 11,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            Cart
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
